src/main/java/appeng/tile/storage/TileChest.java

```diff
@@ -572,87 +579,46 @@ public void onChangeInventory( final IInventory inv, final int slot, final InvOp
 				this.markForUpdate();
 			}
 		}
-	}
-
-	@Override
-	public boolean canInsertItem( final int slotIndex, final ItemStack insertingItem, final EnumFacing side )
-	{
-		if( slotIndex == 1 )
+		if( inv == inputInventory )
 		{
-			if( AEApi.instance().registries().cell().getCellInventory( insertingItem, this, StorageChannel.ITEMS ) != null )
-			{
-				return true;
-			}
-			if( AEApi.instance().registries().cell().getCellInventory( insertingItem, this, StorageChannel.FLUIDS ) != null )
+			if( mc == InvOperation.insert )
 			{
-				return true;
+				this.tryToStoreContents();
 			}
 		}
-		else
-		{
-			try
-			{
-				final IMEInventory<IAEItemStack> cell = this.getHandler( StorageChannel.ITEMS );
-				final IAEItemStack returns = cell.injectItems( AEApi.instance().storage().createItemStack( this.inv.getStackInSlot( 0 ) ), Actionable.SIMULATE,
-						this.mySrc );
-				return returns == null || returns.getStackSize() != insertingItem.getCount();
-			}
-			catch( final ChestNoHandler ignored )
-			{
-			}
-		}
-		return false;
 	}
 
-	@Override
-	public boolean canExtractItem( final int slotIndex, final ItemStack extractedItem, final EnumFacing side )
-	{
-		return slotIndex == 1;
-	}
-
-	@Override
-	public int[] getAccessibleSlotsBySide( final EnumFacing side )
+	protected IItemHandler getItemHandlerForSide( @Nonnull EnumFacing side )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1161)